apps/fxc-front/src/app/workers/score-track.ts

```diff
@@ -0,0 +1,75 @@
+import {BRecord, IGCFile} from 'igc-parser';
+import {ScoreInfo, Scoring, scoringRules, solver} from 'igc-xc-score';
+import {RuntimeTrack} from "@flyxc/common";
+
+export interface Request {
+	track: RuntimeTrack;
+	league?: string;
+	config?: { [key: string]: any } | undefined;
+}
+
+export interface Response {
+	scoreInfo: ScoreInfo;
+	scoring: Scoring;
+	trackId: string;
+}
+
+const w: Worker = self as any;
+
+w.onmessage = (message: MessageEvent<Request>) => {
+	console.info('scoring: received', message);
+	try {
+		const request = message.data;
+		const flight = igcFile(request.track);
+		const solutions = solver(flight, getScoringRules(request), request.config);
+		let solution = solutions.next();
+		if (solution.value.scoreInfo){
+			let response: Response = {
+				scoreInfo: solution.value.scoreInfo,
+				scoring: {name:solution.value.opt.scoring.name, code: solution.value.opt.scoring.code, multiplier: solution.value.opt.scoring.multiplier},
+				trackId: message.data.track.id
+			}
+			console.info("scoring: computed", response.scoreInfo.score, response)
+			w.postMessage(response, {});
+		}
+	} catch (e) {
+		console.error('solver failed', e);
+	}
+};
+
+function getScoringRules(request:Request) {
```

Comment:
Maybe this should be revamped so that flyxc also uses the scoring rules?

What we should think about:
- Do we want to depend on `igc-xc-score` (ixs) rules -> probably not,
- Can we create rules in flyxc that easily convert to ixs

Reply:
For the moment, we have still a dependency to some ixs rules, but they could be easily be removed by copying the remaining ones (XContest, FAI-OAR, FFVL)